doc/spec_TransferElement.dox

```diff
@@ -263,8 +263,8 @@ This documnent is currently still **INCOMPLETE**!
 - \anchor transferElement_C_2 2. In doPostXxx no new ChimeraTK::runtime_error or ChimeraTK::logic_error are thrown. Exceptions that were risen in doPreXxx or doXxxTransferYyy, or that were received from the queue (see B.8.3) are rethrown. \ref transferElement_comment_C_2 "(*)"
   - 2.1 The TransferElement base class stores the caught exceptions in TransferElement::_activeException.
   - 2.2 postXxx() is re-throwing the exception after delegating to doPostXxx().
-  - \anchor transferElement_C_2_3 2.3 If and only if _activeException == true, decorators must swap the _activeException with its target before delegating to the target's postXxx(). This swaps the
-    active exception into the target, which is now responsible for handling it, and swaps in a nullptr from the target, so the decorator level does not throw any more \ref transferElement_comment_C_2_3a "(*)". \ref transferElement_comment_C_2_3b "(*)"
+  - \anchor transferElement_C_2_3 2.3 Decorators must delegate the TransferElement::_activeException to their target using TransferElement::setActiveException(). If the delegated exception is not nullptr,
+  the targets _active exception will be replaced by it, and the delegated exeption, which is handed by reference, is set to nullptr \ref transferElement_comment_C_2_3a "(*)". The target is now responsible for handling the exception. If the delegated exception already is nullptr, setActiveException() has no effect. A nullptr does not overwrite an active exception in the target \ref transferElement_comment_C_2_3b "(*)". \ref transferElement_comment_C_2_3c "(*)"
```

Comment:
Shouldn't there be an exception that the exception is not delegated if it was thrown by the decorator itself?

Reply:
(btw: It is helpful if we have in the source code one line per paragraph. There is an extra newline character here. This makes it harder to read if that newline does not accidentally match roughly the automatic line wrapping of the editor or the github display...)

Reply:
Yes, that's 2.3.1, just below.

Reply:
oh, it still says swap there. I'll fix that

Reply:
I just fixed 2.3.1 and removed the newline. Did not notice it in my browser. Must have aligned perfectly with the line break.